contracts/src/v0.8/shared/token/ERC20/BurnMintERC20.sol

```diff
@@ -0,0 +1,205 @@
+
+  /// @inheritdoc ERC20Burnable
+  /// @dev Uses OZ ERC20 _burn to disallow burning from address(0).
+  /// @dev Decreases the total supply.
+  function burn(uint256 amount) public override(IBurnMintERC20, ERC20Burnable) onlyBurner {
+    super.burn(amount);
+  }
+
+  /// @inheritdoc IBurnMintERC20
+  /// @dev Alias for BurnFrom for compatibility with the older naming convention.
+  /// @dev Uses burnFrom for all validation & logic.
+  function burn(address account, uint256 amount) public virtual override {
+    burnFrom(account, amount);
+  }
+
+  /// @inheritdoc ERC20Burnable
+  /// @dev Uses OZ ERC20 _burn to disallow burning from address(0).
+  /// @dev Decreases the total supply.
+  function burnFrom(address account, uint256 amount) public override(IBurnMintERC20, ERC20Burnable) onlyBurner {
+    super.burnFrom(account, amount);
+  }
+
+  /// @inheritdoc IBurnMintERC20
+  /// @dev Uses OZ ERC20 _mint to disallow minting to address(0).
+  /// @dev Disallows minting to address(this)
+  /// @dev Increases the total supply.
+  function mint(address account, uint256 amount) external override onlyMinter validAddress(account) {
+    if (i_maxSupply != 0 && totalSupply() + amount > i_maxSupply) revert MaxSupplyExceeded(totalSupply() + amount);
+
+    _mint(account, amount);
+  }
+
+  // ================================================================
+  // │                            Roles                             │
+  // ================================================================
+
+  /// @notice grants both mint and burn roles to `burnAndMinter`.
+  /// @dev calls public functions so this function does not require
+  /// access controls. This is handled in the inner functions.
+  function grantMintAndBurnRoles(address burnAndMinter) external {
+    grantRole(MINTER_ROLE, burnAndMinter);
+    grantRole(BURNER_ROLE, burnAndMinter);
+  }
+
+  /// @notice Returns the current CCIPAdmin
+  function getCCIPAdmin() external view returns (address) {
+    return s_ccipAdmin;
+  }
+
+  /// @notice Transfers the CCIPAdmin role to a new address
+  /// @dev only the owner can call this function, NOT the current ccipAdmin, and 1-step ownership transfer is used.
+  /// @param newAdmin The address to transfer the CCIPAdmin role to. Setting to address(0) is a valid way to revoke
+  /// the role
+  function setCCIPAdmin(address newAdmin) public onlyRole(DEFAULT_ADMIN_ROLE) {
+    address currentAdmin = s_ccipAdmin;
+
+    s_ccipAdmin = newAdmin;
+
+    emit CCIPAdminTransferred(currentAdmin, newAdmin);
+  }
+
+  // ================================================================
+  // │                            Access                            │
+  // ================================================================
+
+  /// @notice Checks whether the msg.sender is a permissioned minter for this token
+  /// @dev Reverts with a SenderNotMinter if the check fails
+  modifier onlyMinter() {
```

Comment:
Maybe we can inline these modifiers? Single use modifiers are always bad but even the twice used is not really required I think